### ISSUE TITLE
concept-deepin-topBar (like concept cars) ... ???

### ISSUE THREAD

**Issue by aiamuzz** [kirigayakazushin/deepin-topbar#30]
Hi,

I am a Brand & Design consultant, needless to say Deepin OS caught my attention ... as a professional i can't help but think of 'Deepin' as a brand ... what follows is a natural consequence ... elegance being the keyword ... Deepin is elegant !!!

I am not a coder(i am a Computer Science Graduate and an ex-banker), had i been one(a coder) probably you'd have `deepin-topbar` on the following lines, unfortunately i am not ... so i am open to collaboration with coder.

And If Deepin officially decided to develop the topBar it would almost certainly be like this i guess, i say this as a designer and brand consultant ... the proof of which you can see in this concept topBar mockup ... the elegance of deepin is consistent ... don't you think.

here you go ... for your-eyes-only ... it is NOT a topBar in flesh and blood ... it is just a concept topBar(at least until a coder is willing to collaborate and want to go down this road and realize it) ... just like concept Cars !!! ... 

![deepin_topbar_screen 1](https://user-images.githubusercontent.com/3283676/33237845-01ffdf4a-d2a5-11e7-8923-06c9af38b98d.png)

![deepin_topbar_screen 2](https://user-images.githubusercontent.com/3283676/33237847-05168026-d2a5-11e7-8fdb-fb049c887194.png)

![deepin_topbar_screen 3](https://user-images.githubusercontent.com/3283676/33237848-094392b0-d2a5-11e7-8b5a-971315a7cf65.png)

![deepin_topbar_screen 4](https://user-images.githubusercontent.com/3283676/33237850-0cc799a4-d2a5-11e7-9502-9eaa7727c718.png)

![deepin_topbar_screen 5](https://user-images.githubusercontent.com/3283676/33237852-1012bfe4-d2a5-11e7-8ff9-e39e370dfaa3.png)

![deepin_topbar_screen 6](https://user-images.githubusercontent.com/3283676/33237853-129ba4d8-d2a5-11e7-88a5-16c2d0009e97.png)

![deepin_topbar_screen 7](https://user-images.githubusercontent.com/3283676/33237856-1b0772b4-d2a5-11e7-82ac-30ab9771843a.png)

**Comment by aiamuzz** [comment]
Info : 

I am a lazy person ... as a consequence i would want to pull of things in the least of effort possible ... being a CS graduate the nerd in me can't help but think on the lines ... if i was a coder ... and a buff for efficiency(a lazy person that i am) ...

I thinking out loud ... the following ... 

My Starting point for proj_topBar ... would be deepin-dock ...

that explains ... Screen 1 & Screen 2 ... they are Deepin Dock in Fashion mode and Efficient mode as is.

Screen 3 is the path i would pursue first ... the Lazy_CSgrad_Design_Brand guy that i am ... In this screen you'll see

a Lazy guy takes an existing deepin-dock ... rename it to deepin-dock-topbar ... add a third 'Mode' ... which i call 'Deepin (Unofficial)' ...

Deepin (Unofficial) mode 
... equals ...
`Fashion Mode = true (with the location bottom)` and `Efficient Mode = true (with the location top)`

If this is achieved then we have a topBar to start with ... BUT ... the Efficient top bar of deepin-dock needs to be modified ... if we modify right away then we mess the 'Efficient Mode' ... so i would create a copy(a duplicate) of the Efficient Mode Module within deepin-dock-topbar

so effectively at this point deepin-dock-topbar fork would have 3 modules instead of 2 ... it would have 'Fashion Mode Module'
'Efficient Mode Module'
'Deepin (Unofficial) Mode Module' - which is the duplicate of Efficient Mode Module ... aka ... topbar module

the next step would be ... tweak/develop ... the Deepin (Unofficial) Mode Module ...

If we remove the ability to add icons to the efficient mode in the Deepin (Unofficial) Mode Module ... then what we would be left with is a Deepin-topBar !!!

Screens 4, 5, 6, 7 are ways in which i would develop the Deepin (Unoffcial) topbar ...

**Comment by aiamuzz** [comment]
If the deepin-dock cannot be manipulated as per what i hope to then ... then one will have no choice but to start building a topbar from the ground up ... and Its highly NECESSARY to develop a deepin-topbar on the lines of Screens 4, 5, 6, 7 ... don't you think ?

Deepin is catching up fast for its eye-candy design ... the fact that its dde is being integrated into a lot of other popular distros is reason enough to highly focus on a topbar which is in line with the 'design definition' of deepin OS !!!

@kirigayakazushin ... i have borrowed deepin OS's current design definition and recreated/developed this deepin-topbar roadmap ... i would be willing to assist you in any way i can ... i have a much more detailed roadmap (checklist) to take this project forward ... please let me know.

If its done right then this project could be presented to Deepin Official team for an official consideration to merge into its official deepin-dock ... as it is in line with their universal design definition/philosophy !!!

**Comment by aiamuzz** [comment]
any design feedback is welcome ... and if @kirigayakazushin permits than i would like to continue tweaking the design as per your feedback and post the same here ... but this will be a virtual topbar, only for your eyes ... a concept bar ... just like a concept car !!!

**Comment by kirigayakazushin** [comment]
Thanks for your advices, the company has now decided not to develop the topbar, so I developed it myself. I have done similar to your current design, but technically not very good to achieve, and some of the design does not meet expectations, it will only retain the menu. My principle is to make the best use of existing technologies and data without making any major changes to the system.

**Comment by kirigayakazushin** [comment]
My next plan is that the menu contains only a few shortcuts and there will be an advanced setup program that controls more detailed settings.

**Comment by aiamuzz** [comment]
Hello ... @kirigayakazushin 

That's sad ... If the company has decided not to develop topbar ... and glad you are trying to develop one ... even i think a topbar option is a good one to have.

Like i said i am not a coder ... but i have some questions if you can clarify ...

1. As you are a coder and you know to read the code, is it possible to enable the following in deepin-dock package ?

Fashion Mode = true (with the location bottom)
Efficient Mode = true (with the location top

basically can both fashion mode and efficient mode be switched on at the same time ?

In your free time could you look into the deepin-dock code ?

**Comment by kirigayakazushin** [comment]
This needs to be changed dde-dock, because the content is determined by the product department.And dde-dock is currently only allowed to start one. @aiamuzz

**Comment by aiamuzz** [comment]
i know it is currently allowed to start one ... but if one had to tried ... is it possible ... unless the product department does not permit to do so ... 

2. My second question ... Is it possible to extract/export the entire 'Efficient Mode Module' from the dde-dock package ... if so can it be run as a separate package ?

**Comment by kirigayakazushin** [comment]
In fact, dde-dock also has a backend program that holds and supplies all the data it needs, and the dock is a singleton, so it's not easy to do that.

**Comment by aiamuzz** [comment]
hmmmmm ... that is bad news ...

If somehow one could isolate and export/duplicate/copy the efficient mode to a separate package ... it will be a great way to develop the topbar with minimal changes to the system. Basically i am looking if we can use the current efficient mode implementation as a top bar ...

the topbar you have developed in this project ... how have you approached it ... have you built it from scratch ? or you have built over existing implementation ?

**Comment by kirigayakazushin** [comment]
deepin-topbar is actually from dde-dock. Initially, I used all the frame code directly. In the subsequent submission, I replaced a lot of code to realize the function I wanted. You can think of topbar as an efficient mode dock. I've reimplemented some plug-in interfaces, and based on this added new features.

**Comment by aiamuzz** [comment]
oh ok ... that's good ... i wouldn't understand the technical terminology as i don't understand it completely ... i can understand it only as a concept.

can you reuse the Efficient Mode code from dde-dock and repackage it to run as a top bar ?

the reason i am really hoping this direction be attempted, is because you can get the entire 'system tray' implementation from the efficient-mode implement ... the topbar will get a ready-made system tray with its associated rules ... the proportions icon sizes, icon display rules its menu's, etc ...

If the above can be achieved, removing the app icons from it would be the next thing to do inside the code. Until such a thing is achieved we can remove all icons from this and use it as a standby topbar ...

**Comment by aiamuzz** [comment]
Hey that's great ... so topbar is actually the efficient mode dock ?

Have you modified system-tray bits of code by any chance ?

**Comment by kirigayakazushin** [comment]
@aiamuzz  Yes, but the dock's tray is a plug-in, and the topbar's tray is a module. They are different. You can remove the dock's tray by removing the plugin, but you can not do that for topbar.

**Comment by aiamuzz** [comment]
by dock's tray do you mean ... the default deepin fashion mode dock at the bottom  or  do you mean the system tray(set of icons) seen on the right of the efficient mode bar ?

**Comment by smold22** [comment]
![deepin_top-bart_desktop_1](https://user-images.githubusercontent.com/18718946/36945256-3a7505c6-1fbc-11e8-8d09-e8ba9af52eb2.png)
![deepin_top-bart_desktop_2](https://user-images.githubusercontent.com/18718946/36945257-3a9abea6-1fbc-11e8-851b-79cd635529a2.png)
![deepin_top_bar2](https://user-images.githubusercontent.com/18718946/36945259-52dc69ec-1fbc-11e8-9306-f7e0a006d912.png)
![deepin_top_bar3](https://user-images.githubusercontent.com/18718946/36945260-53020b84-1fbc-11e8-856d-ddcfb36e64c1.png)
![deepin_top_bar4](https://user-images.githubusercontent.com/18718946/36945261-53270fba-1fbc-11e8-97f5-a1facff8cc50.png)
![deepin_top_bar5](https://user-images.githubusercontent.com/18718946/36945262-534cdfce-1fbc-11e8-8e5b-24aa3a64facd.png)
![deepin_top_bar6](https://user-images.githubusercontent.com/18718946/36945263-5372c31a-1fbc-11e8-87a3-69f3147010ad.png)
![deepin_top_bar7](https://user-images.githubusercontent.com/18718946/36945264-539839f6-1fbc-11e8-9eb6-e24096ff099b.png)
![deepin_top_bar8](https://user-images.githubusercontent.com/18718946/36945265-53bde89a-1fbc-11e8-8efa-d0eb892a3ce4.png)
If the appearance and functionality will be similar, it will be quite convenient

**Comment by kirigayakazushin** [comment]
@smold22  Thank you for your suggestion, I will consider it.

**Comment by smold22** [comment]
And why you don't want to try to make a topbar with dynamic blur, remove the date and leave only time. A a date display at the touch of the right mouse button

**Comment by smold22** [comment]
and could you tweak the display of the language selection icon, and instead put the flag of the selected language

**Comment by smold22** [comment]
Here is there is remarkable icons
https://github.com/stephenhutchings/typicons.font.git
![deepinscreenshot_ - _20180325151810](https://user-images.githubusercontent.com/18718946/37873108-c3a6b4fa-303f-11e8-9d81-c30907730140.png)
